### PR TITLE
fix(content-block-media): alignment between web components & react

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
#5534

### Description
Aligned Content Block Media for both Web Components and React, cleaning up and changing the knob structure as well as some miscellaneous spacing fixes.

### Changelog

**Changed**

- changed knob structure and cleanup for both versions
- some spacing changes for Content Block variation components

### CODE REVIEW

**Comment by DragosRistici** [comment]
Hey, @IgnacioBecerra, thank you for taking your time to go over this one. A lot cleaner now. A few little things to fix, please.

This is the reference file:

Screenshot 2021-04-26 at 13 48 41

So,
1. The subheading, or the intro below the heading should be $expressive-paragraph-01. That's 32px/40px/PlexSansLight on max. Now its incorrect:
Screenshot 2021-04-26 at 14 02 25
2. The spacing on top of the $expressive-heading-04 is part of the clean-up and should go from 96px to 64px. Right now, we got a 32px coming from above and another 96px (6 rem) added from below. As you can see, the 96px goes as margin-top AND margin-bottom. Both should be 64px/4rem, not 6rem.
Screenshot 2021-04-26 at 13 55 33
Screenshot 2021-04-26 at 13 55 46
Screenshot 2021-04-26 at 13 55 54
3. Also, for some reason, as I pointed out before, the text links, in content and in lists, for the Gray 100 theme,  drop to 14px in font size, instead of 16px:
Screenshot 2021-04-26 at 13 51 58
Screenshot 2021-04-26 at 13 51 47

4. Kind of like on the same note, the image caption goes to 12px instead of staying at 14 on the Gray 100 theme. 
Screenshot 2021-04-26 at 13 52 52

5. The React has some 16px left and right margins for some reason for the feature card component:
Screenshot 2021-04-26 at 15 23 15

Screenshot 2021-04-26 at 15 21 52

6. React knobs for the Default story has inversed block heading with group heading
Screenshot 2021-04-26 at 15 28 55
Screenshot 2021-04-26 at 15 28 50

7. Knobs for the Link list story are just for the Link list. The others are gone. I believe if you find the elements on the page, the knobs should be there, too. You just add the extra knobs. 

8. Link list story on the React Storybook has the block heading misaligned, also the same problem goes to the Link list component 
Screenshot 2021-04-26 at 15 36 04
Screenshot 2021-04-26 at 15 36 00

9. Container bottom border rule should be by default present, and a knob in every story. Right now it doesn't exist on the default story, as an object or as a knob.

**Comment by IgnacioBecerra** [comment]
@DragosRistici Thanks for the feedback! The g100 styles were fixed in a separate PR (#5887), and the Link List story grid misalignment should be fixed in #5733.
All the other request changes have been implemented. 👍

**Comment by DragosRistici** [comment]
Thank you, @IgnacioBecerra , looks great!

**Review by DragosRistici — APPROVED** [review]

**Comment by jeffchew** [comment]
Hey @IgnacioBecerra , you may need to update snapshots.

File: packages/web-components/src/components/card-link/card-link.scss [hunk 1]
```diff
@@ -14,6 +14,10 @@
 :host(#{$dds-prefix}-card-link),
 :host(#{$dds-prefix}-link-list-item-card),
 :host(#{$dds-prefix}-link-list-item-card-cta) {
+  .#{$prefix}--card {
+    max-width: 320px;
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/card-link/card-link.scss:18]: this max-width will break the specs at larger breakpoints 
Screen Shot 2021-04-27 at 3 18 37 PM
Screen Shot 2021-04-27 at 3 19 50 PM

Comment (jeffchew) [reply packages/web-components/src/components/card-link/card-link.scss]: @IgnacioBecerra can you take a look?

Comment (IgnacioBecerra) [reply packages/web-components/src/components/card-link/card-link.scss]: @ariellalgilmore Good catch, I'm checking with designers about defining a `max-width` for different breakpoints that could fix this. Thanks!

Comment (ariellalgilmore) [reply packages/web-components/src/components/card-link/card-link.scss]: thanks @IgnacioBecerra !!

Comment (IgnacioBecerra) [reply packages/web-components/src/components/card-link/card-link.scss]: fyi: @DragosRistici has confirmed a max-width of 320px has approved by design. PR will be merged shortly :)

**Review by ariellalgilmore — APPROVED** [review]
LGTM! Thanks for confirming the changes!

**Pull request merged**